### PR TITLE
♻️ [timeago/date-countdown/date-display] Split component files for packaging

### PR DESCRIPTION
**Pull request by rcebulko**
~Look only at last 3 commits ([amp-timeago](https://github.com/ampproject/amphtml/pull/32440/commits/066196bbad1b20263452d1a3432110b9ddfc7f8e), [amp-date-countdown](https://github.com/ampproject/amphtml/pull/32440/commits/592873ccc146a5ec52b060635b6dd0cab126f7aa), [amp-date-display](https://github.com/ampproject/amphtml/pull/32440/commits/42928cbc0fec18e0629b9605ed54174954171b91)) until #32407 is merged and I can rebase~

### CODE REVIEW

File: extensions/amp-date-countdown/1.0/base-element.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {DateCountdown} from './component';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {parseDateAttrs as parseDateAttrsBase} from '../../../src/utils/date';
+
+export class BaseElement extends PreactBaseElement {}
```
Comment (jridgewell) [inline extensions/amp-date-countdown/1.0/base-element.js:21]: Wondering what would the `render` interface be like in CE mode without the mustache template?

File: extensions/amp-date-display/1.0/base-element.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {DateDisplay} from './component';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {parseDateAttrs as parseDateAttrsBase} from '../../../src/utils/date';
+
+export class BaseElement extends PreactBaseElement {}
```
Comment (jridgewell) [inline extensions/amp-date-display/1.0/base-element.js:21]: Ditto.

File: extensions/amp-timeago/1.0/base-element.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Timeago} from './component';
+import {parseDateAttrs as parseDateAttrsBase} from '../../../src/utils/date';
+
+export class BaseElement extends PreactBaseElement {}
```
Comment (jridgewell) [inline extensions/amp-timeago/1.0/base-element.js:21]: I think we need the `props['placeholder']` code, too?

Comment (rcebulko) [reply extensions/amp-timeago/1.0/base-element.js]: I think you're right! Done.

File: extensions/amp-date-countdown/1.0/base-element.js (hunk 1 above)
Comment (rcebulko) [reply extensions/amp-date-countdown/1.0/base-element.js:21]: I'll be honest, I'm not familiar enough with the components to to know how to answer this

File: extensions/amp-date-countdown/1.0/amp-date-countdown.js [hunk 4]
```diff
@@ -14,19 +14,17 @@
...
 import {dict} from '../../../src/utils/object';
 import {isExperimentOn} from '../../../src/experiments';
 import {isLayoutSizeDefined} from '../../../src/layout';
-import {parseDateAttrs as parseDateAttrsBase} from '../../../src/utils/date';
 
 /** @const {string} */
 const TAG = 'amp-date-countdown';
 
-class AmpDateCountdown extends PreactBaseElement {
+class AmpDateCountdown extends BaseElement {
```
Comment (rcebulko) [inline extensions/amp-date-countdown/1.0/amp-date-countdown.js:27]: I might need to implement `isLayoutSupported` in `BaseElement`; not 100% sure but I think that's part of the Bento side (`userAssert`/experiment checking is not ofc)

Comment (rcebulko) [reply extensions/amp-date-countdown/1.0/amp-date-countdown.js:27]: Done; please verify

Comment (rcebulko) [reply extensions/amp-date-countdown/1.0/amp-date-countdown.js:27]: Removed per offline convo

**Pull request merged**